src/eval/execution.rs

```diff
@@ -42,7 +70,64 @@ pub fn evaluate(nix_expr: &str) -> EvalResult {
     }
 
     let namespace_obj = module.get_module_namespace().to_object(scope).unwrap();
-    nix_value_from_module(scope, &namespace_obj)
+
+    let nix_value_attr = v8::String::new(scope, "default").unwrap();
+    let Some(nix_value) = namespace_obj.get(scope, nix_value_attr.into()) else {
+        todo!(
+            "Could not find the nix value: {:?}",
+            namespace_obj.to_rust_string_lossy(scope)
+        )
+    };
+    let nix_value: v8::Local<v8::Function> =
+        nix_value.try_into().expect("Nix value must be a function.");
+
+    Ok(nix_value)
+}
+
+fn import_nix_module<'s>(
+    scope: &mut HandleScope<'s>,
+    args: v8::FunctionCallbackArguments<'s>,
+    mut ret: v8::ReturnValue,
+) {
+    let module_path = args.get(0).to_rust_string_lossy(scope);
+    let module_source_str = std::fs::read_to_string(module_path).unwrap();
+
+    let nix_fn = eval_nix_fn_from_string(scope, &module_source_str);
+
+    let nix_fn = match nix_fn {
+        Ok(nix_fn) => nix_fn,
+        Err(err) => {
+            let err_str = v8::String::new(scope, &err).unwrap();
+            let err_obj = v8::Exception::error(scope, err_str);
+            ret.set(err_obj);
+            return;
+        }
+    };
+
+    ret.set(nix_fn.into());
+}
+
+fn exec_module<'a>(
+    code: &str,
+    scope: &mut v8::HandleScope<'a>,
+) -> Result<Local<'a, Object>, String> {
+    let source = to_v8_source(scope, code, "<eval string>");
+    let module = v8::script_compiler::compile_module(scope, source).unwrap();
+
+    if module
+        .instantiate_module(scope, resolve_module_callback)
+        .is_none()
+    {
+        return Err("Instantiation failure.".to_owned());
+    }
+
+    if module.evaluate(scope).is_none() {
+        return Err("Evaluation failure.".to_owned());
+    }
+
+    let obj = module.get_module_namespace().to_object(scope).unwrap();
+
+    Ok(obj)
```

Comment:
Inline obj?
```suggestion
    Ok(module.get_module_namespace().to_object(scope).unwrap())
```

Reply:
Personally I often lean towards breaking up lines into variables to be more descriptive, e.g. if I add error handling in the future, I'd need to replace the .unwrap with something else and the line will grow further.